src/components/msg/utils.tsx

```diff
@@ -456,23 +474,31 @@ export const getMessageByType = (message: any, viewRaw: boolean, t:any) => {
   }
 
   return {
-    type: <Tag
-      value={t(`message_labels:${results.tagDisplay}`)}
-      theme={results.tagTheme}
-    />,
-    message: <results.content message={message as any} />,
+    type: (
+      <Tag
+        value={t(`message_labels:${results.tagDisplay}`)}
+        theme={results.tagTheme}
+      />
+    ),
+    message: <results.content message={message} />,
   };
 };
 
 export const convertMsgsToModels = (transaction: any) => {
   const messages = R.pathOr([], ['messages'], transaction).map((msg, i) => {
     const model = getMessageModelByType(msg?.['@type']);
-    if (model === MODELS.MsgWithdrawDelegatorReward
-      || model === MODELS.MsgWithdrawValidatorCommission) {
+    if (
+      model === MODELS.MsgWithdrawDelegatorReward
+      || model === MODELS.MsgWithdrawValidatorCommission
+    ) {
       const log = R.pathOr(null, ['logs', i], transaction);
-      return model.fromJson(msg, log);
+      const data = model.fromJson(msg, log);
+      return data;
     }
-    return model.fromJson(msg);
+    const data = model.fromJson(msg);
+    // IDK why do we need to do this
+    data.type = msg['@type'];
```

Comment:
I think we should also have a `type` field so this shouldn't be needed?

Reply:
yep

Reply:
@jay-dee7 Okay, I made changes and tested them locally by re-building. It's working fine.